.gitlab-ci.yml

```diff
@@ -21,84 +21,112 @@ generate-scripts:
       - install_script_agent6.sh
       - install_script_agent7.sh
 
-test-opensuse-13:
+.test:
   image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/datadog-agent-buildimages/docker_x64:$DATADOG_AGENT_BUILDIMAGES
   tags: ["runner:docker"]
   stage: test
   dependencies: ["generate-scripts"]
+  script:
+    - ./test/dockertest.sh --image "registry.ddbuild.io/images/mirror/${IMAGE}" --script install_script_agent6.sh 
+      --minor-version "${MINOR_VERSION}"
+      --expected-minor-version "${EXPECTED_MINOR_VERSION}" 
+      --flavor "${FLAVOR}" 
+      --old-suse "${OLD_SUSE}" 
+      --apt-url "$APT_URL" 
+      --apt-repo-version "$APT_REPO_VERSION_AGENT6" 
+      --yum-url "$YUM_URL" 
+      --yum-version-path "$YUM_VERSION_PATH_AGENT6" 
+    - ./test/dockertest.sh --image "registry.ddbuild.io/images/mirror/${IMAGE}" --script install_script_agent7.sh
+      --minor-version "${MINOR_VERSION}"
+      --expected-minor-version "${EXPECTED_MINOR_VERSION}"
+      --flavor "${FLAVOR}"
+      --old-suse "${OLD_SUSE}"
+      --apt-url "$APT_URL"
+      --apt-repo-version "$APT_REPO_VERSION_AGENT7"
+      --yum-url "$YUM_URL"
+      --yum-version-path "$YUM_VERSION_PATH_AGENT7"
+  variables:
+    OLD_SUSE: "false"
+
+test-opensuse-13:
+  extends: .test
+  rules:
+    - if: '$CI_PIPELINE_SOURCE == "push"'
   parallel:
     matrix:
       - IMAGE: opensuse/archive:13.2
       - IMAGE: opensuse/archive:13.2
         FLAVOR: datadog-dogstatsd
       - IMAGE: opensuse/archive:13.2
         FLAVOR: datadog-iot-agent
-  script:
-    - ./test/dockertest.sh --image "registry.ddbuild.io/images/mirror/${IMAGE}" --script install_script_agent6.sh --minor-version "${MINOR_VERSION}" --expected-minor-version "${EXPECTED_MINOR_VERSION}" --flavor "${FLAVOR}" --old-suse "true"
-    - ./test/dockertest.sh --image "registry.ddbuild.io/images/mirror/${IMAGE}" --script install_script_agent7.sh --minor-version "${MINOR_VERSION}" --expected-minor-version "${EXPECTED_MINOR_VERSION}" --flavor "${FLAVOR}" --old-suse "true"
+  variables:
+    OLD_SUSE: "true"
 
+test_pinned_version:
+  extends: .test
+  rules:
+    - if: '$CI_PIPELINE_SOURCE == "push"'
+  parallel:
+    # NOTE: to keep the matrix reasonably sized, we don't test everything everywhere
+    # for example, it's ok to test getting specific flavor/minor version only on
+    # CentOS 7 (yum) and Rocky 9 (dnf) and not on all versions of all RHEL derivatives
+    matrix:
+      - IMAGE: centos:centos7
+        MINOR_VERSION: 38
+      - IMAGE: rockylinux:9.0
+        MINOR_VERSION: 38
+      - IMAGE: ubuntu:14.04
+        MINOR_VERSION: 38
+      - IMAGE: debian:10.9
+        MINOR_VERSION: 38
+      # When installing pinned minor version and some repos are broken,
```

Comment:
I would prefer a jira ticket rather than commented code :)

Reply:
This is not a new comment (this was just moved), the corresponding task is AP-1885.

Reply:
I don't necessarily comment only on new code, sorry for that